### PR TITLE
Add a simple codify filter

### PR DESCRIPTION
**Pull request by spapas**
This should implement #1052

### CODE REVIEW

**Comment by lorenzhs** [comment]
What would you say should be the expected behaviour for snippets containing URLs? I think in ``, codify should take precedence over the URL linkification (`irclinky`) and thus be applied *before* it. GitHub also does it that way.

Other than that, I really like it! Thanks a lot!

**Comment by spapas** [comment]
Hello @lorenzhs after some research I found out that the `DOMfilter` filter splits the code containing links to multiple strings thus if a string like "foo \`http://www.bar.baz bar\`" is input the `codify` filter will get two strings as an input: 

foo \`

and 

http://www.bar.baz bar\`

thus it won't work properly :(

A solution I was able to found was to call codify directly right after the `conditionalCodify` without `DOMFilter` i.e the filters will be like

``::part.text | conditionalLinkify:part.classes.includes('cof-chat_host') | codify | DOMfilter:'irclinky'  | ``

Is it ok to proceed like this ?

**Comment by lorenzhs** [comment]
Sorry, I meant before `conditionalLinkify` :\ `irclinky` only does linkification of irc channel names.

Just wrapping things in a `<code>` tag without escaping is no good because it is vulnerable to cross-site scripting. I forgot whether angular handles this, sorry.

**Comment by spapas** [comment]
Hello @lorenzhs maybe i didn't make myself clear so I've pushed my commit with the fix to take a look at. 

There shouldn't be any vulnerabilities to cross-site scripting with the `codify` filter, it just adds the `<code>` tags where needed and passes everything to the other filters so every conversion of non-safe code should still be performed. 

Please take a look at it thank you!

File: js/filters.js [hunk 1]
```diff
@@ -237,4 +237,14 @@ weechat.filter('prefixlimit', function() {
     };
 });
 
+weechat.filter('codify', function() {
+    return function(text) {
+        var re = /(`.+?`)/g;
```
Comment (lorenzhs) [inline js/filters.js]: Why `.+?`? Isn't that equivalent to `.*`?

**Comment by lorenzhs** [comment]
Yeah I think since you don't have any `$sce.trustAsHtml` thingies, it should be okay.

@torhve can you have a look? Is this XSS-safe?

**Comment by torhve** [comment]
I don't have the bandwidth to do full testing, but I think this makes sense, I like this change

**Comment by lorenzhs** [comment]
Thanks a lot, @spapas!

**Pull request merged**